Tests/Reqnroll.FeatureSourceGenerator.Tests/MSTest/MSTestFeatureSourceGeneratorTests.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Reqnroll.FeatureSourceGenerator.MSTest;
+
+public class MSTestFeatureSourceGeneratorTests
+{
+    [Fact]
+    public void GeneratorProducesMSTestOutput()
+    {
+        var references = AppDomain.CurrentDomain.GetAssemblies()
+            .Where(asm => !asm.IsDynamic)
+            .Select(asm => MetadataReference.CreateFromFile(asm.Location));
+
+        var compilation = CSharpCompilation.Create("test", references: references);
+
+        var generator = new CSharpTestFixtureSourceGenerator([BuiltInTestFrameworkHandlers.MSTest]);
+
+        const string featureText =
+            """
+            #language: en
+            @featureTag1
+            Feature: Calculator
+
+            @mytag
+            Scenario: Add two numbers
+                Given the first number is 50
+                And the second number is 70
+                When the two numbers are added
+                Then the result should be 120
+            """;
+
+        var driver = CSharpGeneratorDriver
+            .Create(generator)
+            .AddAdditionalTexts([new FeatureFile("Calculator.feature", featureText)])
+            .RunGeneratorsAndUpdateCompilation(compilation, out var generatedCompilation, out var diagnostics);
+
+        diagnostics.Should().BeEmpty();
+
+        generatedCompilation.SyntaxTrees.Should().HaveCount(1);
+        var generatedSyntaxTree = generatedCompilation.SyntaxTrees.Single().Should().BeAssignableTo<CSharpSyntaxTree>().Subject!;
+
+        generatedSyntaxTree.GetDiagnostics().Should().BeEmpty();
+
+        var generatedSyntaxRoot = (CompilationUnitSyntax)generatedSyntaxTree.GetRoot();
+        generatedSyntaxRoot.ChildNodes().Should().HaveCount(1);
+
+        var generatedNamespace = generatedSyntaxRoot.ChildNodes().Single().Should().BeAssignableTo<NamespaceDeclarationSyntax>().Subject;
+        generatedNamespace.Name.ToString().Should().Be("test");
+
+        generatedNamespace.Members.Should().HaveCount(1);
+        var generatedClass = generatedNamespace.Members.Single().Should().BeAssignableTo<ClassDeclarationSyntax>().Subject;
+
+        generatedClass.AttributeLists.Should().HaveCount(1);
+        var generatedClassAttributes = generatedClass.AttributeLists.Single().Attributes;
+
+        generatedClassAttributes.Should().HaveCount(1);
+        var generatedClassAttribute = generatedClassAttributes.Single();
+
+        generatedClassAttribute.Name.ToString().Should().Be("global::Microsoft.VisualStudio.TestTools.UnitTesting.TestClass");
```

Comment:
All this required to assert:
- There were no diagnostic issues raised during generation
- The generated file contains a class in a "test" namespace 
- The class is decorated with the MSTest attribute

Reply:
If we wanted to check for other attributes, or the test methods were present with _their_ attributes, it's probably triple the line-count.

I do prefer this to checking an exact string output, but it's significantly harder to read and I think presents an immediate maintenance headache.

Reply:
I recently discovered the `Subject` / `Which` properties of fluent assertions that can make this code slightly better:

```
generatedClass.AttributeList.Should().ContainSingle()
  .Which.Should().ContainSingle()
  .Which.Name.ToString().Should().Be("global::Microsoft.VisualStudio.TestTools.UnitTesting.TestClass");
```

Reply:
`Should().ContainSingle()` does help reduce the bloat. This could just be a case of needing some syntactic sugar to make the statements less hurtful.

```
var generatedClass = generatedSyntaxRoot.Should().ContainNamespaceDeclaration("test")
    .Which.Should().ContainClassDeclaration("Calculator");

generatedClass.Should().HaveAttribute("global::Microsoft.VisualStudio.TestTools.UnitTesting.TestClass");
```